cmd/podman/generate/spec.go

```diff
@@ -0,0 +1,65 @@
+package pods
+
+import (
+	"fmt"
+
+	"github.com/containers/podman/v4/cmd/podman/common"
+	"github.com/containers/podman/v4/cmd/podman/registry"
+	"github.com/containers/podman/v4/cmd/podman/utils"
+	"github.com/containers/podman/v4/pkg/domain/entities"
+	"github.com/spf13/cobra"
+)
+
+var (
+	specCmd = &cobra.Command{
+		Use:               "spec [options] {CONTAINER... |POD...}",
+		Short:             "Generate Specgen JSON based on containers or pods",
+		Long:              "Generate Specgen JSON based on containers or pods",
+		RunE:              spec,
+		Args:              cobra.ExactArgs(1),
+		ValidArgsFunction: common.AutocompleteContainers,
+		Example:           `podman generate spec ctrID`,
+	}
+)
+
+var (
+	opts *entities.GenerateSpecOptions
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Command: specCmd,
+		Parent:  generateCmd,
+	})
+	opts = &entities.GenerateSpecOptions{}
+	flags := specCmd.Flags()
+
+	filenameFlagName := "filename"
+	flags.StringVarP(&opts.FileName, filenameFlagName, "f", "", "Write output to the specified path")
+
+	verboseFlagName := "verbose"
+	flags.BoolVarP(&opts.Verbose, verboseFlagName, "v", true, "print the file contents to Stdout")
+
+	compactFlagName := "compact"
+	flags.BoolVarP(&opts.Compact, compactFlagName, "c", false, "print the json in a compact format for consumption")
+
+	cloneFlagName := "clone"
+	flags.BoolVar(&opts.Clone, cloneFlagName, false, "create a cloned name so that the spec can be used without modifications")
```

Comment:
Should this just be --name?  Rather then podman generating a random name?

Reply:
sure, that works for me